runtime/tests/checker/access_test.go

```diff
@@ -95,47 +108,112 @@ func TestCheckAccessModifierCompositeFunctionDeclaration(t *testing.T) {
 			ast.AccessPublicSettable: false,
 		}
 
-		require.Len(t, tests, len(ast.Accesses))
+		require.Len(t, tests, len(ast.BasicAccesses))
 
 		for access, expectSuccess := range tests {
-			for _, isInterface := range []bool{true, false} {
 
-				interfaceKeyword := ""
-				body := ""
-				if isInterface {
-					interfaceKeyword = "interface"
+			testName := fmt.Sprintf(
+				"%s/%s",
+				compositeKind.Keyword(),
+				access.Keyword(),
+			)
+
+			t.Run(testName, func(t *testing.T) {
+
+				_, err := ParseAndCheck(t,
+					fmt.Sprintf(
+						`
+                          %[1]s Test {
+                              %[2]s fun test() {}
+                          }
+	                    `,
+						compositeKind.Keyword(),
+						access.Keyword(),
+					),
+				)
+
+				if expectSuccess {
+					assert.NoError(t, err)
 				} else {
-					body = "{}"
+					errs := ExpectCheckerErrors(t, err, 1)
+
+					assert.IsType(t, &sema.InvalidAccessModifierError{}, errs[0])
```

Comment:
```suggestion
					expectInvalidAccessModifierError(t, err)
```

Reply:
Not sure if this is necessary (it kind of hides details about the test), but figured I'd suggest anyways

Reply:
Great idea! I've looked for reusable expectation functions and fixed them all up in f347e69a1af1d42c10180f285eb7e85cda200351